opm/grid/cpgrid/CpGridData.hpp

```diff
@@ -257,6 +285,592 @@ class CpGridData
+                            boundary_patch_faces.push_back(face_idx);
+                            // Associate each old face on the boundary of the patch with the new born ones.
+                            boundary_old_to_new_faces.push_back(std::make_tuple(face_idx, children_list));
+                        }
+                    } // end k-for-loop
+                } // end i-for-loop
+            } // end j-for-loop
+            // To store the relation between parent cell and its new-born-cells.
+            // {parent index (coarse grid), {child 0 index, child 1 index, ... (refined grid)}}
+            std::vector<std::tuple<int,std::vector<int>>> parent_to_children_cells;
+            parent_to_children_cells.reserve(patch_dim[0]*patch_dim[1]*patch_dim[2]);
+            // To store the relation between a new-born-cell and its parent cell.
+            std::vector<std::array<int,2>> child_to_parent_cells; // {child index (refined grid), parent cell index (coarse grid)}
+            child_to_parent_cells.reserve(xfactor*yfactor*zfactor);
+            for (int k = 0; k < grid_dim[2]; ++k) {
+                for (int j = 0; j < grid_dim[1]; ++j) {
+                    for (int i = 0; i < grid_dim[0]; ++i) {
+                        int cell_idx = (k*grid_dim[0]*grid_dim[1]) + (j*grid_dim[0]) +i;
+                        std::vector<int> children_list;
+                        if ( (i > startIJK[0]-1) && (i < endIJK[0]) && (j > startIJK[1]-1) && (j < endIJK[1])
+                             && (k > startIJK[2]-1) && (k < endIJK[2])) {
+                            for (int n = (k-startIJK[2])*cells_per_dim[2]; n < (k-startIJK[2]+1)*cells_per_dim[2]; ++n) {
+                                for (int m = (j-startIJK[1])*cells_per_dim[1]; m < (j-startIJK[1]+1)*cells_per_dim[1]; ++m) {
+                                    for (int l = (i-startIJK[0])*cells_per_dim[0]; l < (i-startIJK[0]+1)*cells_per_dim[0]; ++l) {
+                                        children_list.push_back((n*xfactor*yfactor) + (m*xfactor) + l);
+                                        child_to_parent_cells.push_back({(n*xfactor*yfactor) + (m*xfactor) + l, cell_idx});
+                                    }// end l-for-loop
+                                } // end m-for-loop
+                            } // end n-for-loop
+                            parent_to_children_cells.push_back(std::make_tuple(cell_idx, children_list));
+                        }// end if 'patch cells'
+                    } // end i-for-loop
+                } // end j-for-loop
+            } // end k-for-loop
+            // Map to identify parent faces. {index face, true/false}. True when a face got refined.
+            std::map<int,bool> isParent_faces;
+            for (int face = 0; face < this-> face_to_cell_.size(); ++face) {
+                isParent_faces[face] = false;
+            }
```

Comment:
We should use a std::vector<int> for performance and memory reasons:
```c++
std::vector<int> isParent_faces(face_to_cell_.size(), false);
for (auto& face : patch_faces) {
  isParent_faces[face] = true;
}
```

Reply:
When
`std::vector<int> isParent_faces(face_to_cell_.size(), false);`
does it count as "reserve" or "resize"? The way info is stored in the next lines depends on that.